node/src/blockchain/blockchain_interface/blockchain_interface_web3/mod.rs

```diff
@@ -577,7 +653,8 @@ mod tests {
                     },
                 ]
             }
-        )
+        );
+        TestLogHandler::new().exists_log_containing(&format!("DEBUG: {test_name}: Retrieving transactions {expected_log} for: 0x3f69…72fc chain_id: 137 contract: 0xee9a352f6aac4af1a5b9f467f6a93e0ffbe9dd35"));
```

Comment:
I'd prefer that you wouldn't insist on asserting on these logs. That's for in general. 

Here, it's even more clumsy, because your test performs this five times (with each call of assert_on_retrieves_transactions). It means that after the first call we only can't ever fail because if it worked once it will appease each attempt by which we check on the presence of this string in the list of printed logs. 

Again, I'd prefer to see that you deleted it wholly. There isn't much or almost any use at all in it.